### PR TITLE
Minimax

### PR DESCRIPTION
**Pull request by vandyliu**

### CODE REVIEW

File: c4.pl [hunk 1]
```diff
@@ -35,19 +46,117 @@
...
+playMove(Depth, Alpha, Beta, Colour, Board, Move, Score) :-
+    \+ Depth = 0,
+    newBoard(Board, Move, Colour, NewBoard),
+    (win(NewBoard) -> 
+    (maxPlayer(Colour) ->
+        Score is (1 + Depth) ; % Quicker wins are worth more
+        Score is (-1 - Depth)) ;
+        getAvailableMoves(NewBoard, NewAvailableMoves),
+        (NewAvailableMoves == [] ->
+            Score is 0;
```
Comment (EricJ111) [inline c4.pl:135]: why is score 0 in this case? isn't it the outcome of getBestScore?

**Comment by EricJ111** [comment]
idk about u but i ran out of stack space after ~20 total moves

**Comment by EricJ111** [comment]
ur fuckin cracked at prolog tho

**Comment by vandyliu** [comment]
> idk about u but i ran out of stack space after ~20 total moves

did you `set_prolog_flag(stack_limit, 8_147_483_648).`
For me, I can usually play a full game even without this^

File: c4.pl (hunk 1 above)
Comment (vandyliu) [reply c4.pl:135]: In OOP, this translates to
```
if noMoreMovesAvailable (tie-game) {
  Score = 0 // tie
} else {
  NewDepth = Depth - 1
  playMoves(...) // Keep trying other moves
  getBestScore(..) // Get best score out of above moves
}
```

Comment (EricJ111) [reply c4.pl:135]: oh i didn't see the if case gotcha

**Comment by EricJ111** [comment]
i set that flag and it fixed it nice

**Review by EricJ111 — APPROVED** [review]
gs

**Pull request merged**